### PR TITLE
Fix for #329 in echocat/puppet-graphite

### PR DESCRIPTION
**Pull request by arymkus**
PR to incorporate previously published patch which generates proper config for carbon-cache instances.
See #329 for more details.

### CODE REVIEW

**Comment by arymkus** [comment]
@dwerder BTW, I'll have some spare time next week and can fix linter-related warnings.
Please reply if you want me to do it.

**Comment by dwerder** [comment]
That would be nice.

**Comment by arymkus** [comment]
Done.

**Comment by dwerder** [comment]
my only question is, why you want to set the pidfile in the carbon.conf, because it is already set in each init file. And the path is:  scope.lookupvar('graphite::storage_dir_REAL')

https://github.com/echocat/puppet-graphite/blob/master/templates/etc/init.d/RedHat/carbon-cache.erb#L25

**Comment by arymkus** [comment]
Because carbon itself has ignored the higher scope's PATH variable in it's config and still stored it's .pid files in /var/run/... (or wherever the dafault path is) but the properly generated systemd's unit files contained new path to .pid files.
So the final behavior was - systemd starts carbon (and it was running well) but wasn't able to read actual carbon's .pid file and marked the service as failed. That was leading to manifest's failure.

File: templates/opt/graphite/conf/carbon.conf.erb [hunk 1]
```diff
@@ -32,6 +32,10 @@
 #
 LOCAL_DATA_DIR = <%= scope.lookupvar('graphite::local_data_dir_REAL') %>/
 
+<% unless [:undef, nil].include? scope.lookupvar('graphite::gr_storage_dir') -%>
+PID_DIR = <%= scope.lookupvar('graphite::gr_storage_dir') %>
```
Comment (dwerder) [inline templates/opt/graphite/conf/carbon.conf.erb]: This has to be scope.lookupvar('graphite::storage_dir_REAL')  to match the init scripts.

**Comment by arymkus** [comment]
in case of either storage_dir or whisper_dir param used - carbon creates its .pid files in correct place.
And only with local_data_dir carbon ignored proper location.
I've tested my fix and it works.

**Comment by arymkus** [comment]
ping

**Pull request merged**